crates/wasmtime/src/runtime/values.rs

```diff
@@ -475,6 +482,20 @@ impl From<Option<Rooted<AnyRef>>> for Val {
     }
 }
 
+impl From<Rooted<StructRef>> for Val {
+    #[inline]
+    fn from(val: Rooted<StructRef>) -> Val {
+        Val::AnyRef(Some(val.cast()))
```

Comment:
Above I was thinking that there should be `unchecked_cast` to require more scrutiny, but now I'm wondering if it might be good to have that *and* specific methods per-type to cast "safely" to the parent types? Not memory-safety-related-safety but more correctness-related-safety. That way the upcast method could use `unchecked_cast` internally but `unchecked_cast` wouldn't be used externally much as the safer upcasts would be used instead.

Reply:
How are you imagining this? Because this `From` impl *is* that safe upcast in my imagination. We could have a trait or something tho too

```rust
pub trait Upcast<T> {}

impl Rooted<T> {
    pub fn upcast<U>(self) -> Rooted<U>
    where
        T: Upcast<U>
    {
        // ...
    }
}
```

Reply:
(we would have to seal the trait as well)

Reply:
Oh I'm thinking something like:

```rust
impl Rooted<StructRef> {
    fn to_anyref(&self) -> Rooted<AnyRef> { self.unchecked_cast() }
}
```

and repeating that for `Rooted<...>` as appropriate and having `to_foo` method for anything that's appropriate

Reply:
Ah, gotcha. Yeah I can do that.